index.js

```diff
@@ -75,7 +111,30 @@ Babel.prototype.baseDir = function() {
 };
 
 Babel.prototype.transform = function(string, options) {
-  return transpiler.transform(string, options);
+  if (transformIsParallelizable(options)) {
+    // send the job to the worker pool
+    return new Promise(function(resolve, reject) {
```

Comment:
Can we extract this to its own file, so we can test this branch in isolation? Parallel/worker stuff is